src/ciflow-bot.ts

```diff
@@ -0,0 +1,220 @@
+import * as probot from 'probot';
+
+export default function runCIFlowBot(app: probot.Application): void {
+  new CIFlowBot(app).main();
+}
+
+export class CIFlowBot {
+  // Constructor required
+  readonly app: probot.Application;
+
+  // Configurations
+  readonly allowed_command: string[] = ['ciflow'];
+  readonly bot_app_name = 'pytorchbot';
+  readonly bot_assignee = 'pytorchbot';
+  readonly pr_label_prefix = 'ciflow/';
+  readonly rollout_users = ['zhouzhuojie']; // slow rollout to specific group of users first
+
+  // Stateful instance variables
+  command = '';
+  command_args: string[] = [];
+  comment_author = '';
+  comment_body = '';
+  ctx: probot.Context = null;
+  dispatch_labels: string[] = [];
+  dispatch_strategies: string[] = ['add_default_labels'];
+  event = '';
+  owner = '';
+  pr_author = '';
+  pr_labels: string[] = [];
+  pr_number = 0;
+  repo = '';
+
+  constructor(app: probot.Application) {
+    this.app = app;
+  }
+
+  valid(): boolean {
+    if (this.event !== 'pull_request' && this.event !== 'issue_comment') {
+      this.ctx.log.error({ctx: this.ctx}, 'Unknown webhook event');
+      return false;
+    }
+
+    if (this.event === 'issue_comment') {
+      if (this.comment_author === '') {
+        this.ctx.log.error({ctx: this.ctx}, 'Empty comment author');
+        return false;
+      }
+
+      // only the pr author can trigger new changes to the ciflow
```

Comment:
This sounds unreasonably restrictive, often project members will adjust CI flags for 1st time contributors

Reply:
will relax this for the next PR. the check for member's write permission requires additional logic that's nice to test and commit separately - I added a TODO for this